subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheToolingApiInvocationIntegrationTest.groovy

```diff
@@ -99,6 +102,74 @@ class ConfigurationCacheToolingApiInvocationIntegrationTest extends AbstractConf
         outputContains("Reusing configuration cache.")
     }
 
+    def "configuration cache is disabled for model building actions"() {
```

Comment:
We should probably test each of the cases where the cache should be disabled: requesting a model from the client, running a build action, and running a phased build action.